core/src/main/java/com/scalar/db/transaction/consensuscommit/Coordinator.java

```diff
@@ -106,7 +122,20 @@ Optional<Coordinator.State> getStateForGroupCommit(String fullId) throws Coordin
       return stateContainingTargetTxId;
     }
 
-    return get(createGetWith(fullId));
+    return getStateByIdOnly(fullId);
+  }
+
+  /**
+   * Gets the coordinator state only by ID, without considering the parent child IDs. The scope of
+   * this method is public, but it's only for internal use.
+   *
+   * @param id the ID of the coordinator state
+   * @return the coordinator state
+   * @throws CoordinatorException if the coordinator state cannot be retrieved
+   */
+  public Optional<Coordinator.State> getStateByIdOnly(String id) throws CoordinatorException {
+    Get get = createGetWith(id);
+    return get(get);
   }
```

Comment:
I feel that the already exposed getState() method behaves the same way when the ID is not a full ID for the coordinator group commit.
So, we need this method purely for performance reasons, without checking the ID format. Is that correct?

Reply:
@brfrn169 Sorry for lack of explanation. The current implementation doesn't allow external components to directly look up a state by the parent ID for the coordinator group commit. It's sometimes useful for optimization (e.g., for caching coordinator state records with parent IDs as keys) in external components.

I realized the method names introduced in this PR were a bit unclear from the use case perspective. I updated them in https://github.com/scalar-labs/scalardb/pull/2612/commits/496c8f8ab487804bdfdf6c68fd38cf0435dce6f6. I hope it clarifies the code.